src/analyses/constant_propagator.cpp

```diff
@@ -204,19 +113,16 @@ void constant_propagator_domaint::transform(
     exprt g;
 
     if(from->get_target()==to)
-      g = simplify_expr(from->guard, ns);
+      g=simplify_expr(from->guard, ns);
     else
-      g = simplify_expr(not_exprt(from->guard), ns);
+      g=simplify_expr(not_exprt(from->guard), ns);
 
-    if (g.is_false())
+    if(g.is_false())
      values.set_to_bottom();
     else
     {
-      //TODO: we need to support widening!
-      if (g.is_constant())
-        values.set_to_top();
-      else
-        two_way_propagate_rec(g, ns);
+      two_way_propagate_rec(g, ns);
+      assert(!values.is_bottom); // for now
```

Comment:
I think the two way propagation (which handles the variable propagation through gotos, e.g. [a == 3] goto ...) is disabled in this commit (the body of two_way_propagate_rec()) is commented out). Therefore propagating through the goto can never yield bottom which the assertion checks for. When two way propagation is enabled again in the future the assertion will not hold anymore.